### DISCUSSION TITLE
Model crashes while getting Word Embeddings

### DISCUSSION THREAD

**Discussion by moayadeldin** [aub-mind/arabert#163] [Q&A]
When I try to get word embeddings for a dataset using the approach mentioned in [Arabic Embeddings Notebook on Colab](https://colab.research.google.com/drive/1Xk8IGcwGTr-YNlB6lhl8jYdIU_M7Mcjx#scrollTo=hX4hJWDUh1B7) for relatively big input (+100 sentences) then Google Colab completely crashes as it ran out of memory. I tried to use my own PC but it crashed as well. 
Is there any way to solve this problem other than dividing my dataset into very small pieces?
TIA.

**Comment by WissamAntoun** [comment]
try adding a torch no grad context.

```python
with torch.no_grad():
    outputs = model(**inputs)
```

Colab should be able to handle 100 sentences

**Reply by moayadeldin** [reply]
Thank You! I used the same approach to handle much more sentences (+3000) and unfortunately it crashed, is there a way allowing me to handle many thousands of sentences without having this ram crash?

**Reply by WissamAntoun** [reply]
did you move the embeddings to CPU or it's crashing the GPU RAM?

Also, try saving the results on disk

**Reply by moayadeldin** [reply]
It crashes while using either GPU or CPU of Google Colab, and kindly tell me if there is a way to save results on disk while the model is running instead of RAM as notebook crashes before model finishes getting its output.
